pennylane_lightning/src/Util.hpp

```diff
@@ -50,3 +50,10 @@ template<typename T, typename... Args>
 std::unique_ptr<T> make_unique(Args&&... args) {
     return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
 }
+
+// Exception for functions that aren't implemented
+class NotImplementedException : public std::logic_error
+{
+public:
+    NotImplementedException() : std::logic_error("Function not yet implemented") { };
```

Comment:
(Minor): perhaps it could be used even when it's not planned that there is going to be an implementation provided?
```suggestion
    NotImplementedException() : std::logic_error("Function is not implemented.") { };
```